src/schemas/vendor.ts

```diff
@@ -15,4 +15,10 @@ export const vendorSchema = z.object({
   dt: z.string().datetime(),
 })
 
+export const vendorPageSearchSchema = z.object({
+  page: z.number().catch(1),
+  product_name: z.string().catch(''),
+  location: z.string().catch(''),
```

Comment:
Why use catch here? Interesting choice

Reply:
Using `.catch()` here is a way to provide default values if the input isn’t valid. For example, `z.number().catch(1)` ensures that if the input isn’t a valid number, it defaults to 1 instead of throwing an error. It’s a good choice when you prefer defaults over strict validation errors.

Reply:
I followed the tanstack documentation [here](https://tanstack.com/router/v1/docs/framework/react/guide/search-params), saying that:

> In the above example, we used Zod's .catch() modifier instead of .default() to avoid showing an error to the user because we firmly believe that if a search parameter is malformed, you probably don't want to halt the user's experience through the app to show a big fat error message. That said, there may be times that you do want to show an error message. In that case, you can use .default() instead of .catch().